src/Numerics/ODESolvers/AdditiveRungeKuttaMethod.jl

```diff
@@ -137,51 +145,75 @@ mutable struct AdditiveRungeKutta{
         variant_storage = additional_storage(variant, Q, Nstages)
         VS = typeof(variant_storage)
 
-        # The code throughout assumes SDIRK implicit tableau so we assert that
-        # here.
-        for is in 2:Nstages
-            @assert RKA_implicit[is, is] ≈ RKA_implicit[2, 2]
+        implicit_solvers = Dict()
+
+        rk_diag = unique(diag(RKA_implicit))
+        # Remove all zero entries from `rk_diag`
+        # so we build all unique implicit solvers (parameterized by the
+        # corresponding RK coefficient)
+        filter!(c -> !iszero(c), rk_diag)
+
+        # LowStorageVariant ARK methods assume that both the explicit and
+        # implicit B and C vectors are the same. Additionally, the diagonal
+        # of the implicit Butcher table A is assumed to have the form:
+        # [0, c, ... c ], where c is some non-zero constant.
+        if variant isa LowStorageVariant
+            @assert RKB_explicit == RKB_implicit
+            @assert RKC_explicit == RKC_implicit
+            # rk_diag here has been filtered of all non-unique and zero values.
+            # So [0, c, ... c ] filters to [c]
+            if length(rk_diag) != 1
+                error("The implicit Butcher table must have SDIRK form when using LowStorageVariant.")
+            end
```

Comment:
I'd be ok with this being an `@assert`
```suggestion
            @assert length(rk_diag) == 1
```
Since all the others checks above are `@asserts`